### ISSUE TITLE
metadata died during json encoding of response

### ISSUE THREAD

**Issue by aemneina** [rancher/rancher#9637]
**Rancher versions:**
rancher/server: 1.5.5

**Infrastructure Stack versions:**
network-services:
rancher/metadata:v0.9.1

Alena states: some reason json encoding of the response had failed

```
2017-08-08T06:19:04.844372451Z time="2017-08-08T06:19:04Z" level=info msg="Applied http://<rancher_server>/v1/configcontent/metadata-answers?client=v2&requestedVersion=2745391?version=2745391-14a7928f14789d3a00ab33efdfd9c22c" 
2017-08-08T06:19:04.868169386Z time="2017-08-08T06:19:04Z" level=info msg="Download and reload in: 251.074725ms" 
2017-08-08T06:19:04.881180152Z fatal error: scan missed a g
2017-08-08T06:19:04.888464509Z 
2017-08-08T06:19:04.888495330Z goroutine 324061 [running]:
2017-08-08T06:19:04.888499169Z runtime.throw(0xb23be5, 0xf)
2017-08-08T06:19:04.888501879Z 	/usr/local/go/src/runtime/panic.go:566 +0x95 fp=0xc420a3d170 sp=0xc420a3d150
2017-08-08T06:19:04.888504752Z runtime.gcMarkRootCheck()
2017-08-08T06:19:04.888507716Z 	/usr/local/go/src/runtime/mgcmark.go:124 +0x275 fp=0xc420a3d1c0 sp=0xc420a3d170
2017-08-08T06:19:04.888510693Z runtime.gcMarkDone()
2017-08-08T06:19:04.888520101Z 	/usr/local/go/src/runtime/mgc.go:1079 +0xc5 fp=0xc420a3d1e8 sp=0xc420a3d1c0
2017-08-08T06:19:04.888523241Z runtime.gcAssistAlloc(0xc420995a00)
2017-08-08T06:19:04.888525829Z 	/usr/local/go/src/runtime/mgcmark.go:502 +0x369 fp=0xc420a3d270 sp=0xc420a3d1e8
2017-08-08T06:19:04.888529138Z runtime.mallocgc(0x8, 0xa45160, 0x1, 0xc400000004)
2017-08-08T06:19:04.888531813Z 	/usr/local/go/src/runtime/malloc.go:563 +0x9a4 fp=0xc420a3d310 sp=0xc420a3d270
2017-08-08T06:19:04.888534379Z runtime.newobject(0xa45160, 0xc421586200)
2017-08-08T06:19:04.888536944Z 	/usr/local/go/src/runtime/malloc.go:785 +0x38 fp=0xc420a3d340 sp=0xc420a3d310
2017-08-08T06:19:04.888539389Z encoding/json.typeEncoder(0x1005240, 0xa3f9c0, 0x15)
2017-08-08T06:19:04.888543110Z 	/usr/local/go/src/encoding/json/encode.go:333 +0x4f fp=0xc420a3d3d0 sp=0xc420a3d340
2017-08-08T06:19:04.888545808Z encoding/json.valueEncoder(0xa3f9c0, 0x0, 0x15, 0xa3f9c0)
2017-08-08T06:19:04.888548273Z 	/usr/local/go/src/encoding/json/encode.go:328 +0x5c fp=0xc420a3d408 sp=0xc420a3d3d0
2017-08-08T06:19:04.888550615Z encoding/json.(*encodeState).reflectValue(0xc4204fa0b0, 0xa3f9c0, 0x0, 0x15, 0x100)
2017-08-08T06:19:04.888553246Z 	/usr/local/go/src/encoding/json/encode.go:307 +0x3f fp=0xc420a3d440 sp=0xc420a3d408
2017-08-08T06:19:04.888556001Z encoding/json.interfaceEncoder(0xc4204fa0b0, 0xa279a0, 0xc421586200, 0x94, 0xc421450100)
2017-08-08T06:19:04.888558734Z 	/usr/local/go/src/encoding/json/encode.go:573 +0xdb fp=0xc420a3d480 sp=0xc420a3d440
2017-08-08T06:19:04.888561781Z encoding/json.(*mapEncoder).encode(0xc42014c568, 0xc4204fa0b0, 0xa3f9c0, 0xc4205ab140, 0x15, 0xa30100)
2017-08-08T06:19:04.888564463Z 	/usr/local/go/src/encoding/json/encode.go:646 +0x542 fp=0xc420a3d5d8 sp=0xc420a3d480
2017-08-08T06:19:04.888567039Z encoding/json.(*mapEncoder).(encoding/json.encode)-fm(0xc4204fa0b0, 0xa3f9c0, 0xc4205ab140, 0x15, 0xc4205a0100)
2017-08-08T06:19:04.888569704Z 	/usr/local/go/src/encoding/json/encode.go:662 +0x64 fp=0xc420a3d618 sp=0xc420a3d5d8
2017-08-08T06:19:04.888572169Z encoding/json.(*encodeState).reflectValue(0xc4204fa0b0, 0xa3f9c0, 0xc4205ab140, 0x15, 0xc4205a0100)
2017-08-08T06:19:04.888575054Z 	/usr/local/go/src/encoding/json/encode.go:307 +0x82 fp=0xc420a3d650 sp=0xc420a3d618
2017-08-08T06:19:04.888577420Z encoding/json.interfaceEncoder(0xc4204fa0b0, 0xa279a0, 0xc420abe0e0, 0x194, 0xa20100)
2017-08-08T06:19:04.888593403Z 	/usr/local/go/src/encoding/json/encode.go:573 +0xdb fp=0xc420a3d690 sp=0xc420a3d650
2017-08-08T06:19:04.888598976Z encoding/json.(*arrayEncoder).encode(0xc42014c530, 0xc4204fa0b0, 0x9eb140, 0xc420cc7ca0, 0x97, 0x100)
2017-08-08T06:19:04.888603234Z 	/usr/local/go/src/encoding/json/encode.go:723 +0xee fp=0xc420a3d6e8 sp=0xc420a3d690
2017-08-08T06:19:04.888607914Z encoding/json.(*arrayEncoder).(encoding/json.encode)-fm(0xc4204fa0b0, 0x9eb140, 0xc420cc7ca0, 0x97, 0xc4204f0100)
2017-08-08T06:19:04.888612493Z 	/usr/local/go/src/encoding/json/encode.go:730 +0x64 fp=0xc420a3d728 sp=0xc420a3d6e8
2017-08-08T06:19:04.888615829Z encoding/json.(*sliceEncoder).encode(0xc42014c538, 0xc4204fa0b0, 0x9eb140, 0xc420cc7ca0, 0x97, 0x9e0100)
2017-08-08T06:19:04.888623400Z 	/usr/local/go/src/encoding/json/encode.go:697 +0xc1 fp=0xc420a3d768 sp=0xc420a3d728
2017-08-08T06:19:04.888626173Z encoding/json.(*sliceEncoder).(encoding/json.encode)-fm(0xc4204fa0b0, 0x9eb140, 0xc420cc7ca0, 0x97, 0xc420cc0100)
2017-08-08T06:19:04.888628686Z 	/usr/local/go/src/encoding/json/encode.go:709 +0x64 fp=0xc420a3d7a8 sp=0xc420a3d768
2017-08-08T06:19:04.888631121Z encoding/json.(*encodeState).reflectValue(0xc4204fa0b0, 0x9eb140, 0xc420cc7ca0, 0x97, 0x100)
2017-08-08T06:19:04.888633633Z 	/usr/local/go/src/encoding/json/encode.go:307 +0x82 fp=0xc420a3d7e0 sp=0xc420a3d7a8
2017-08-08T06:19:04.888636040Z encoding/json.(*encodeState).marshal(0xc4204fa0b0, 0x9eb140, 0xc420cc7ca0, 0xb20100, 0x0, 0x0)
2017-08-08T06:19:04.888638575Z 	/usr/local/go/src/encoding/json/encode.go:280 +0xb8 fp=0xc420a3d818 sp=0xc420a3d7e0
2017-08-08T06:19:04.888641039Z encoding/json.Marshal(0x9eb140, 0xc420cc7ca0, 0x12, 0xb206a7, 0x9, 0xb218e9, 0xb)
2017-08-08T06:19:04.888643521Z 	/usr/local/go/src/encoding/json/encode.go:145 +0x8f fp=0xc420a3d860 sp=0xc420a3d818
2017-08-08T06:19:04.888645899Z main.respondJSON(0xffb560, 0xc4209ee820, 0xc42071c000, 0x9eb140, 0xc420cc7ca0)
2017-08-08T06:19:04.888648378Z 	/go/src/github.com/rancher/rancher-metadata/main.go:554 +0x39 fp=0xc420a3d8a8 sp=0xc420a3d860
2017-08-08T06:19:04.888652374Z main.respondSuccess(0xffb560, 0xc4209ee820, 0xc42071c000, 0x9eb140, 0xc420cc7ca0)
2017-08-08T06:19:04.888655011Z 	/go/src/github.com/rancher/rancher-metadata/main.go:477 +0xc2 fp=0xc420a3d8e0 sp=0xc420a3d8a8
2017-08-08T06:19:04.888657494Z main.(*ServerConfig).metadata(0xc42015a0e0, 0xffb560, 0xc4209ee820, 0xc42071c000)
2017-08-08T06:19:04.888660086Z 	/go/src/github.com/rancher/rancher-metadata/main.go:439 +0xd39 fp=0xc420a3db98 sp=0xc420a3d8e0
2017-08-08T06:19:04.888662654Z main.(*ServerConfig).(main.metadata)-fm(0xffb560, 0xc4209ee820, 0xc42071c000)
2017-08-08T06:19:04.888665176Z 	/go/src/github.com/rancher/rancher-metadata/main.go:304 +0x48 fp=0xc420a3dbc8 sp=0xc420a3db98
2017-08-08T06:19:04.888667629Z net/http.HandlerFunc.ServeHTTP(0xc420530030, 0xffb560, 0xc4209ee820, 0xc42071c000)
2017-08-08T06:19:04.888670219Z 	/usr/local/go/src/net/http/server.go:1726 +0x44 fp=0xc420a3dbf0 sp=0xc420a3dbc8
2017-08-08T06:19:04.888676644Z github.com/rancher/rancher-metadata/vendor/github.com/gorilla/mux.(*Router).ServeHTTP(0xc420126370, 0xffb560, 0xc4209ee820, 0xc42071c000)
2017-08-08T06:19:04.888679581Z 	/go/src/github.com/rancher/rancher-metadata/vendor/github.com/gorilla/mux/mux.go:99 +0x255 fp=0xc420a3dce0 sp=0xc420a3dbf0
2017-08-08T06:19:04.888682021Z net/http.serverHandler.ServeHTTP(0xc42018cb00, 0xffb560, 0xc4209ee820, 0xc42071c000)
2017-08-08T06:19:04.888684576Z 	/usr/local/go/src/net/http/server.go:2202 +0x7d fp=0xc420a3dd28 sp=0xc420a3dce0
2017-08-08T06:19:04.888687111Z net/http.(*conn).serve(0xc420152300, 0xffc320, 0xc420319c00)
2017-08-08T06:19:04.888689713Z 	/usr/local/go/src/net/http/server.go:1579 +0x4b7 fp=0xc420a3df88 sp=0xc420a3dd28
2017-08-08T06:19:04.888692278Z runtime.goexit()
2017-08-08T06:19:04.888694803Z 	/usr/local/go/src/runtime/asm_amd64.s:2086 +0x1 fp=0xc420a3df90 sp=0xc420a3df88
2017-08-08T06:19:04.888697279Z created by net/http.(*Server).Serve
2017-08-08T06:19:04.888699639Z 	/usr/local/go/src/net/http/server.go:2293 +0x44d
2017-08-08T22:32:40.688663229Z time="2017-08-08T22:32:40Z" level=info msg="Subscribing to events" 
2017-08-08T22:32:40.715637716Z time="2017-08-08T22:32:40Z" level=info msg="Starting rancher-metadata v0.9.1" 
```

**Comment by alena1108** [comment]
Most likely the same issue as https://github.com/golang/go/issues/16083. The fix will be to build using go version 1.8.3

**Comment by deniseschannon** [comment]
Available in v1.6-development with `rancher/metadata:v0.9.4`

**Comment by sangeethah** [comment]
Tested with rancher-srever version - v1.6.8-rc4 and rancher/metadata:v0.9.4

Cattle and K8s automation runs succeeded (except for known issues).
